docs/source/package_reference/builder_classes.mdx

```diff
@@ -0,0 +1,31 @@
+# Builder classes
+
+🤗 Datasets relies on two main classes during the dataset building process: [`datasets.DatasetBuilder`] and [`datasets.BuilderConfig`].
+
+[[autodoc]] datasets.DatasetBuilder
+
+[[autodoc]] datasets.GeneratorBasedBuilder
+
+[[autodoc]] datasets.BeamBasedBuilder
+
+[[autodoc]] datasets.ArrowBasedBuilder
+
+[[autodoc]] datasets.BuilderConfig
+
+[[autodoc]] datasets.DownloadManager
+
+[[autodoc]] datasets.GenerateMode
```

Comment:
Display the table in Markdown in the docstring for `GenerateMode`:

|                                     | Downloads | Dataset |
|-------------------------------------|-----------|---------|
| `REUSE_DATASET_IF_EXISTS` (default) | Reuse     | Reuse   |
| `REUSE_CACHE_IF_EXISTS`             | Reuse     | Fresh   |
| `FORCE_REDOWNLOAD`                  | Fresh     | Fresh   |

Reply:
handled in https://github.com/huggingface/datasets/pull/3690/commits/1ff531cda64d1aa44230c277b5322051312f1816